components/candy-machine/UpdateCandyMachine.tsx

```diff
@@ -178,22 +189,43 @@ const UpdateCandyMachine: FC<{
 
                 state.data.solTreasuryAccount = state.wallet
                 state.data.itemsRedeemed = state.itemsRedeemed
-                setErrorMessage('')
                 return state.data
             } catch (err) {
-                setErrorMessage((err as Error).message)
+                setError((err as Error)?.message)
             }
             setIsLoading(false)
         }
     }
 
     useEffect(() => {
-        setErrorMessage('')
+        setError('')
         setIsLoading(false)
         fetchCandyMachine().then(setCandyMachineConfig)
         setIsLoading(false)
     }, [connection])
 
+    useEffect(() => {
+        const updateInitialState = {
+            price: candyMachineConfig?.price ? new BN(candyMachineConfig?.price).toNumber() / LAMPORTS_PER_SOL : 0,
+            'number-of-nfts': 0,
+            'treasury-account': candyMachineConfig?.solTreasuryAccount?.toBase58() ?? '',
+            captcha: candyMachineConfig?.gatekeeper ?? false,
+            mutable: candyMachineConfig?.isMutable ?? false,
+            'date-mint': candyMachineConfig?.goLiveDate
+                ? parseDateFromDateBN(candyMachineConfig?.goLiveDate)
+                : getCurrentDate(),
+            'time-mint': candyMachineConfig?.goLiveDate
+                ? parseTimeFromDateBN(candyMachineConfig?.goLiveDate)
+                : getCurrentTime(),
+
+            storage: '',
+            files: [],
+            cache: null,
+            'new-authority': '',
+        } as const
+        setValues(updateInitialState)
+    }, [candyMachineConfig])
```

Comment:
I think `initialState` already holds all these values, so this is duplicated code?